server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -106,13 +113,19 @@ public void afterRefresh(boolean didRefresh) {
                                 .max(Comparator.comparingLong(IndexFileNames::parseGeneration));
```

Comment:
Could we use:

```
final String latestSegmentInfos = segmentInfos.getSegmentsFileName();
```

Instead of filtering the collection?

Reply:
`segmentInfosFiles` can contain multiple `segments_N` files. We are making sure to get list of all such files and pick only the latest one by generation.